pkg/assembler/grammar.go

```diff
@@ -0,0 +1,199 @@
+package assembler
+
+import (
+	"fmt"
+	"math"
+)
+
+// Grammar and AST
+
+type CasmProgram struct {
+	Instructions []Instruction `@@`
+}
+
+type Instruction struct {
+	Core      *CoreInstruction `@@`
+	ApPlusOne bool             `( "," @"ap" "+" "+" )? ";" |`
```

Comment:
Is `ap++` followed by a comma or a semicolon? The whitepaper seems to use a semicolon; has that changed in recent versions?

Reply:
<expr>,ap++; is the standar in current Casm and Cairo 0.11